src/maingoblint.ml

```diff
@@ -256,7 +253,7 @@ let preprocess_files () =
   let rec preprocess_arg_file = function
     | filename when Filename.basename filename = "Makefile" ->
       let comb_file = MakefileUtil.generate_and_combine filename ~all_cppflags in
-      [basic_preprocess ~all_cppflags comb_file]
+      [(comb_file, basic_preprocess ~all_cppflags comb_file)]
```

Comment:
I don't think this really works for the old Makefile support because having cilly combine the C files doesn't allow us to know what the actual files were. The `comb_file` isn't actually modified by the user, so we'd be oblivious to the changes and would have to rerun cilly anyway.

Reply:
Yes, it is still somewhat broken, but arguably less so then before, where the dependency information as simply empty.

Reply:
Might be worth adding a comment that this doesn't give the original dependency structure, unlike all the other mechanisms.